### ISSUE TITLE
[BUG]<Completed job ends with "Heating disabled by safety timer.">

### ISSUE THREAD

**Issue by RealGeneDoc** [prusa3d/Prusa-Firmware#3298]
Please, before you create a new bug report, please make sure you searched in open and closed issues and couldn't find anything that matches.

**Printer type** - MK3S+
**Printer firmware version**- 3.10.0-4481

**MMU Upgrade** - no

**SD card or USB/Octoprint** - SD Card

**Describe the bug**
I have two MK3S+ printers and both produce the same issue.  With Firmware 3.10.0-4481 and Slicer 2.3.3, my completed jobs now generate a "Heating disabled by safety timer" error.  A common practice for me is to have a job finish and I'm not back to the printer for 8-10 hours after the job finished.  I've used these printers for 18+ months or so and not encountred this issue.  When I've shown up to the printers immediately after the job was completed, the bed heat remains at 60C for quite some time.  I don't know how long it takes for the error to fire.
  
**To Reproduce**
<see above>
  
**Expected behavior**
Job completes and bed heater turns off.
  
**G-code**
[MiniBase.txt](https://github.com/prusa3d/Prusa-Firmware/files/7404701/MiniBase.txt)

**Video**
n/a

**Comment by gudnimg** [comment]
> I don't know how long it takes for the error to fire.

I took a look where the code fails. If you're not using Farm Mode it is 30 minutes. With Farm Mode enabled it is 45 minutes.

![image](https://user-images.githubusercontent.com/8218499/138597250-14ddfde8-fea3-4873-bc8b-5ad5b109f810.png)

**Comment by gudnimg** [comment]
Any chance you can reproduce this on less than 5 minute print and post the gcode here? I’m interested to learn why the heater doesn’t turn off when the print is finished. I have the same printer as you.

**Comment by neophyl** [comment]
Your attached gcode has M107 to turn the fan off, M104 S0 to turn the extruder off but does not appear to have a M140 S0 command to turn the bed heating off.  It does have an M190 S60 to heat up the bed and wait for temp to be reached.

As a Prusa Slicer appends a copy of the settings to the end of the gcode its not in there either.  So the question is, does whatever printer profile you are using in PS have the gcode commands added ?

**Comment by RealGeneDoc** [comment]
Here's a tiny print that does reproduce the problem.
[OneSmallSquare.txt](https://github.com/prusa3d/Prusa-Firmware/files/7425685/OneSmallSquare.txt)

**Comment by neophyl** [comment]
Once again that gcode has no commands to turn the bed heating off after a print.  So its exactly correct that the printer doesn't turn the bed off until the safety timer kicks in, its not been told to.  

You need to 'fix' your printer profile in whatever slicer you are using (PS2.3.3 from the header).  There is nothing wrong with the printer firmware.  Save one of your projects (file >save as) and then zip up the resulting 3mf file and attach it here.  That will allow someone to check what you actually have configured against the printer in Printer>Custom Gcode

According to the settings appended to the gcode you have Prusa Slicer configured for the following - 
Printer Type ="- default FFF -" this printer also has almost no start/end gcode configured.  Just the real basics
Filament is = "Prusament PLA - Clean1"
Print profile ="40GyroidNoBrimSupport"

So it appears you don't have a Mk3 printer selected and as this is the git to report issues with a MK2/3 printer it kind of makes sense to make sure you are actually configured for one of those before reporting an issue. 

Once you attach a PS 3mf project file someone can see if you have profile issues.  It  may mean a clean install of PS is required.

**Comment by RealGeneDoc** [comment]
My scenario is that I've been using my printers for nearly two years.  Nonetheless, I have been only using fairly vanilla settings with only very minor tweaks.  I have simply been upgrading the software over time, so I didn't even realize the profile of the printer type had been lost from the initial install years ago.  I was directed to post here from the Prusa forum based on the problems I was having.  I can go back and look at the printer type setting, as I didn't intentionally change most settings after downloading the software from Prusa.  I'll go back and ensure it's set up for an MK3S+ and see if that fixes the issue.  Thanks

Edit:  Ok.  It clearly was my fault and I appreciate folks help.  With previous updates, my settings were preserved.  I missed that with a recent update some of my settings were lost.  When I went back through the configuration wizard, I see the M140 S0 command (as well as many other commands) were added to the GCode.  Thanks again
